
# <i>Okay, so you're Brad Pitt:</i><br><br><center>An Analysis of Movie Revenues based on Actors and Movie Genres</center>


## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#wrangling">Data Wrangling</a></li>
<li><a href="#eda">Exploratory Data Analysis</a></li>
<li><a href="#conclusions">Conclusions</a></li>
</ul>

<a id='intro'></a>
## Introduction

As English-American actor Bob Hope put it,

> <i>Celebrities have a way of touching our lives. Perhaps we are influenced by their screen image, or perhaps by their acquired status [...]</i> <

Celebrities have a certain charm, some more than others, which makes us want to watch them. In the context of movies, this can - but not necessarily - translate to higher revenues for films and the companies that produced them. This "charm" aspect of actors and its effect on the marketability and financial success of movies has been the topic in academic literature for many years.

The empirical analysis undertaken in this project aims to contribute to this discussion by examining the TMDB dataset containing data on movies from 1960 to 2017. In particular, this analysis aims to find out if a correlation exists between (top) stars and movie revenues. Another goal of this project is to show (or attempt to show) if an interplay between the actors, genres, and the invested budget correlates with the commercial success of movies.
<br><br><br>


In [1]:
# Import packages for the analysis
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

<a id='wrangling'></a>
## Data Wrangling

To start the analysis, the TMDB dataset needs to be loaded in a Pandas dataframe on which cleaning methods will be applied.

### General Properties

In [3]:
# load the dataset in a dataframe and show the first five rows

main_df = pd.read_csv(r"C:\Users\49176\Desktop\Udacity\Data Analyst Nanodegree\Project 2 - Investigate a Dataset\tmdb-movies.csv", header = 0, index_col = 'id' )
main_df.head()

imdb_id  popularity     budget     revenue  \
id                                                     
135397  tt0369610   32.985763  150000000  1513528810   
76341   tt1392190   28.419936  150000000   378436354   
262500  tt2908446   13.112507  110000000   295238201   
140607  tt2488496   11.173104  200000000  2068178225   
168259  tt2820852    9.335014  190000000  1506249360   

                      original_title  \
id                                     
135397                Jurassic World   
76341             Mad Max: Fury Road   
262500                     Insurgent   
140607  Star Wars: The Force Awakens   
168259                     Furious 7   

                                                     cast  \
id                                                          
135397  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
76341   Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
262500  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
140607  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
168259  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   

                                                 homepage          director  \
id                                                                            
135397                      http://www.jurassicworld.com/   Colin Trevorrow   
76341                         http://www.madmaxmovie.com/     George Miller   
262500     http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   
140607  http://www.starwars.com/films/star-wars-episod...       J.J. Abrams   
168259                           http://www.furious7.com/         James Wan   

                              tagline  \
id                                      
135397              The park is open.   
76341              What a Lovely Day.   
262500     One Choice Can Destroy You   
140607  Every generation has a story.   
168259            Vengeance Hits Home   

                                                 keywords  \
id                                                          
135397  monster|dna|tyrannosaurus rex|velociraptor|island   
76341    future|chase|post-apocalyptic|dystopia|australia   
262500  based on novel|revolution|dystopia|sequel|dyst...   
140607              android|spaceship|jedi|space opera|3d   
168259                car race|speed|revenge|suspense|car   

                                                 overview  runtime  \
id                                                                   
135397  Twenty-two years after the events of Jurassic ...      124   
76341   An apocalyptic story set in the furthest reach...      120   
262500  Beatrice Prior must confront her inner demons ...      119   
140607  Thirty years after defeating the Galactic Empi...      136   
168259  Deckard Shaw seeks revenge against Dominic Tor...      137   

                                           genres  \
id                                                  
135397  Action|Adventure|Science Fiction|Thriller   
76341   Action|Adventure|Science Fiction|Thriller   
262500         Adventure|Science Fiction|Thriller   
140607   Action|Adventure|Science Fiction|Fantasy   
168259                      Action|Crime|Thriller   

                                     production_companies release_date  \
id                                                                       
135397  Universal Studios|Amblin Entertainment|Legenda...       6/9/15   
76341   Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15   
262500  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15   
140607          Lucasfilm|Truenorth Productions|Bad Robot     12/15/15   
168259  Universal Pictures|Original Film|Media Rights ...       4/1/15   

        vote_count  vote_average  release_year    budget_adj   revenue_adj  
id                                                                          
135397        5562           6.5          2015  1.379999e+08  1.392446e+09  
76341         6185       

In [4]:
# Show the number of rows and columns in the dataframa
main_df.shape

(10866, 20)

The dataset contains **10866** rows with **20** columns. The dataset was loaded with the column $\mathit{id}$ as the index.

In [6]:
# Show the data types of each column
main_df.dtypes

imdb_id                  object
popularity              float64
budget                    int64
revenue                   int64
original_title           object
cast                     object
homepage                 object
director                 object
tagline                  object
keywords                 object
overview                 object
runtime                   int64
genres                   object
production_companies     object
release_date             object
vote_count                int64
vote_average            float64
release_year              int64
budget_adj              float64
revenue_adj             float64
dtype: object

We see here that the there are a lot of columns (11, to be exact) which are of type $\mathit{object}$. These are mostly text (string) fields about the movies in our dataset. It is important to know these fields as data manipulation largely relies on the data type.

In [7]:
# Show string fields of the dataframe
main_df.select_dtypes(include = 'object')

imdb_id                original_title  \
id                                                
135397  tt0369610                Jurassic World   
76341   tt1392190            Mad Max: Fury Road   
262500  tt2908446                     Insurgent   
140607  tt2488496  Star Wars: The Force Awakens   
168259  tt2820852                     Furious 7   
...           ...                           ...   
21      tt0060371            The Endless Summer   
20379   tt0060472                    Grand Prix   
39768   tt0060161           Beregis Avtomobilya   
21449   tt0061177        What's Up, Tiger Lily?   
22293   tt0060666      Manos: The Hands of Fate   

                                                     cast  \
id                                                          
135397  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
76341   Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
262500  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
140607  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
168259  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
...                                                   ...   
21      Michael Hynson|Robert August|Lord 'Tally Ho' B...   
20379   James Garner|Eva Marie Saint|Yves Montand|Tosh...   
39768   Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...   
21449   Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...   
22293   Harold P. Warren|Tom Neyman|John Reynolds|Dian...   

                                                 homepage            director  \
id                                                                              
135397                      http://www.jurassicworld.com/     Colin Trevorrow   
76341                         http://www.madmaxmovie.com/       George Miller   
262500     http://www.thedivergentseries.movie/#insurgent    Robert Schwentke   
140607  http://www.starwars.com/films/star-wars-episod...         J.J. Abrams   
168259                           http://www.furious7.com/           James Wan   
...                                                   ...                 ...   
21                                                    NaN         Bruce Brown   
20379                                                 NaN  John Frankenheimer   
39768                                                 NaN      Eldar Ryazanov   
21449                                                 NaN         Woody Allen   
22293                                                 NaN    Harold P. Warren   

                                                  tagline  \
id                                                          
135397                                  The park is open.   
76341                                  What a Lovely Day.   
262500                         One Choice Can Destroy You   
140607                      Every generation has a story.   
168259                                Vengeance Hits Home   
...                                                   ...   
21                                                    NaN   
20379   Cinerama sweeps YOU into a drama of speed and ...   
39768                                                 NaN   
21449                           WOODY ALLEN STRIKES BACK!   
22293        It's Shocking! It's Beyond Your Imagination!   

                                                 keywords  \
id                                                          
135397  monster|dna|tyrannosaurus rex|velociraptor|island   
76341    future|chase|post-apocalyptic|dystopia|australia   
262500  based on novel|revolution|dystopia|sequel|dyst...   
140607              android|spaceship|jedi|space opera|3d   
168259                car race|speed|revenge|suspense|car   
...                                                   ...   
21                               surfer|surfboard|surfing   
20379                           car race|racing|formula 1   
39768                            car|trolley|stealing car   
21449                            

As stated in this <a href='https://s3.amazonaws.com/video.udacity-data.com/topher/2018/July/5b57919a_data-set-options/data-set-options.pdf'>document</a>, some columns contain multiple values which are separated by delimiters. This can be observed in the columnds $\textit{asad}, \textit{asad}, \textit{asad}$

> **Tip**: You should _not_ perform too many operations in each cell. Create cells freely to explore your data. One option that you can take with this project is to do a lot of explorations in an initial notebook. These don't have to be organized, but make sure you use enough comments to understand the purpose of each code cell. Then, after you're done with your analysis, create a duplicate notebook where you will trim the excess and organize your steps so that you have a flowing, cohesive report.

> **Tip**: Make sure that you keep your reader informed on the steps that you are taking in your investigation. Follow every code cell, or every set of related code cells, with a markdown cell to describe to the reader what was found in the preceding cell(s). Try to make it so that the reader can then understand what they will be seeing in the following cell(s).

### Data Cleaning (Replace this with more specific notes!)

In [ ]:
# After discussing the structure of the data and any problems that need to be
#   cleaned, perform those cleaning steps in the second part of this section.


<a id='eda'></a>
## Exploratory Data Analysis

> **Tip**: Now that you've trimmed and cleaned your data, you're ready to move on to exploration. Compute statistics and create visualizations with the goal of addressing the research questions that you posed in the Introduction section. It is recommended that you be systematic with your approach. Look at one variable at a time, and then follow it up by looking at relationships between variables.

### Research Question 1 (Replace this header name!)

In [ ]:
# Use this, and more code cells, to explore your data. Don't forget to add
#   Markdown cells to document your observations and findings.


### Research Question 2  (Replace this header name!)

In [ ]:
# Continue to explore the data to address your additional research
#   questions. Add more headers as needed if you have more questions to
#   investigate.


<a id='conclusions'></a>
## Conclusions

> **Tip**: Finally, summarize your findings and the results that have been performed. Make sure that you are clear with regards to the limitations of your exploration. If you haven't done any statistical tests, do not imply any statistical conclusions. And make sure you avoid implying causation from correlation!

> **Tip**: Once you are satisfied with your work, you should save a copy of the report in HTML or PDF form via the **File** > **Download as** submenu. Before exporting your report, check over it to make sure that the flow of the report is complete. You should probably remove all of the "Tip" quotes like this one so that the presentation is as tidy as possible. Congratulations!